src/main/java/net/sf/jabref/logic/logging/GuiAppender.java

```diff
@@ -37,10 +39,25 @@ public static GuiAppender createAppender(@PluginAttribute("name") String name,
         return new GuiAppender(name, filter, layout);
     }
 
-
+    /**
+     * The log event will be forwarded to the {@link LogMessages} archive.
+     * <p/>
+     * All logged messages will pass this method. For each line in the messages of the log event a new log event will be created.
+     * @param event log event
+     */
     @Override
     public void append(LogEvent event) {
-        CACHE.add(new String(this.getLayout().toByteArray(event)));
+        String message = new String(this.getLayout().toByteArray(event));
+        message = message.replace("\n\r", "\n").replace("\r", "\n");
+        // stack traces logged by 'Log.error("message"), e' will be split by new lines so we can create a new log event for each line as 'e.printStackTrace()' would do.
+        if (event.getLevel() == Level.ERROR) {
+            Arrays.asList(message.split("\n")).stream().filter(s -> !s.isEmpty()).forEach(log -> {
+                LogEvent messageWithPriority = Log4jLogEvent.newBuilder().setMessage(new SimpleMessage(log.replace("\n", ""))).setLevel(event.getLevel()).build();
```

Comment:
I see. What happens if you use setText(logMessage.getMessage().getFormattedMessage()) to show the message instead of setText(logMessage.getMessage().toString()) ?